toolkit/compare_case_groups/api.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2024 Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project.
+#
+
+
+import pandas as pd
+import polars as pl
+
+from toolkit.compare_case_groups.build_dataframes import (
+    build_attribute_df,
+    build_grouped_df,
+    build_ranked_df,
+    filter_df,
+)
+from toolkit.compare_case_groups.temporal_process import (
+    build_temporal_data,
+    create_window_df,
+)
+
+
+class CompareCaseGroups:
+    model_df = pl.DataFrame()
+    filtered_df = pl.DataFrame()
+    final_df = pl.DataFrame()
+
+    def __init__(self):
+        self.filters = []
+        self.groups = []
+        self.aggregates = []
+        self.temporal = ""
+
+    def get_dataset_proportion(self) -> int:
+        initial_row_count = len(self.model_df)
+        filtered_row_count = len(self.filtered_df)
+        return round(
+            100 * filtered_row_count / initial_row_count
+            if initial_row_count > 0
+            else 0,
+            0,
+        )
+
+    def _select_columns_ranked_df(self, ranked_df: pl.DataFrame) -> None:
+        lower_groups = [g.lower() for g in self.groups]
+
+        if self.temporal != "":
+            columns = [
+                *lower_groups,
+                "group_count",
+                "group_rank",
+                "attribute_value",
+                "attribute_count",
+                "attribute_rank",
+                f"{self.temporal}_window",
+                f"{self.temporal}_window_count",
+                f"{self.temporal}_window_rank",
+                f"{self.temporal}_window_delta",
+            ]
+        else:
+            columns = [
+                *lower_groups,
+                "group_count",
+                "group_rank",
+                "attribute_value",
+                "attribute_count",
+                "attribute_rank",
+            ]
```

Comment:
what about:

```python
columns = [g.lower() for g in self.groups]

columns.extend(["group_count", "group_rank", "attribute_value", "attrubite_count", "attribute_rank"]  # You could move this into a constant :)

if self.temporal:
    columns.extend([f"{self.temporal}_window", ...])


```

that way you avoid having the else statement and avoid some duplicated lines

Reply:
Yes, Thanks!